modules/tinymce/src/core/main/ts/html/Base64Uris.ts

```diff
@@ -22,7 +22,7 @@ export interface Base64UriParts {
 }
 
 export const extractBase64DataUris = (html: string): Base64Extract => {
-  const dataImageUri = /data:[^;]+;base64,([a-z0-9\+\/=\s]+)/gi;
+  const dataImageUri = /data:[^;<]+;base64,([a-z0-9\+\/=\s]+)/gi;
```

Comment:
This still is far too open... take for example this HTML:
```
<img data-custom="data:" src="data:image/png;base64,R0/yw==">
```
The above regex will incorrectly walk across the attribute boundaries, e.g: `data:" src="data:image/png;base64,R0/yw==`. So at the very least we need to make sure that spaces aren't allowed in the mime-type part of this regex either. Ideally, we should do some quick investigation if we can get something that will only match a valid mime-type (though this could be difficult, so doing the space might be all we can do).

Reply:
this regex `/data:([a-z0-9]+\/[a-z0-9-.+]+);base64,([a-z0-9\+\/=\s]+)/gi` works if we only concern about matching mime-type formats. I prefer this since it is more precise

Reply:
Except that doesn't match everything that's allowed in a mime type... take a look at the RFC: https://datatracker.ietf.org/doc/html/rfc6838#section-4.2. You can see it allows other things such as `!`, `&`, etc...  which wouldn't be matched by the above.

Reply:
It would open a can of worm if we are going to support what the spec says. I am just going to update the current regex to handle more cases for now